Work with HTTP API v2

tomchiverton opened the pull request:
## What did you implement:

Closes #47 
## How did you implement it:

Copy v2 props to what seems to be right place in the old v1 event

## How can we verify it:

```
service: name-here

provider:
  name: aws
  runtime: nodejs12.x

  httpApi:
    payload: '2.0'
    cors: true

functions:
  dist:
    handler: handler.dist
    events:
      - httpApi:
          path: /dist/{file_name}
          method: get
```
```
const path = require('path');

const StaticFileHandler = require('serverless-aws-static-file-handler')
const clientFilesPath = path.join(__dirname,"./dist/")
const fileHandler = new StaticFileHandler(clientFilesPath);

module.exports.dist = async (event,context) => {
  if (!event.rawPath.startsWith("/dist/")) {
    throw new Error(`[404] Invalid filepath for this resource: ${event.rawPath}`)
  }

  return fileHandler.get(event, context);
}
```

## Todos:

- [x] Write tests
- [x] Fix linting errors
- [ ] All tests pass on our CI Server: https://travis-ci.org/activescott/serverless-aws-static-file-handler

activescott commented:
Thanks for the PR @tomchiverton . If you run `npm run lint-fix` it will usually fix up the failing lint item checks here. I think if you push here again the checks will run immediately. I'll try to keep an eye out for a faster look next time too. Thanks again for the PR!!

tomchiverton commented:
Done

Workflow still waiting approval though.

activescott commented:
@tomchiverton You can ignore the note about travis-ci.org, instead just make sure that the checks pass on the PR itself (which are really just run in github actions https://github.com/activescott/serverless-aws-static-file-handler/actions).

I'm pretty sure adding a unit test would pull the coverage back up above the threshold. Let me know if you have any questions or want a hand.

activescott commented:
I also found some setting on the repo to loosen up the restrictions on running the github actions. Hopefully it won't require an approval for you every time now.

tomchiverton commented:
Any pointers on how to set up a new unit test for this? 
-- 
†øღ
Sent from a super computer that fits in my pocket and is connected to the sum total of all human knowledge 

On 29 December 2021 18:02:54 GMT, scott willeke ***@***.***> wrote:
***@***.*** You can ignore the note about travis-ci.org, instead just make sure that the checks pass on the PR itself (which are really just run in github actions https://github.com/activescott/serverless-aws-static-file-handler/actions).
>
>I'm pretty sure adding a unit test would pull the coverage back up above the threshold. Let me know if you have any questions or want a hand.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/activescott/serverless-aws-static-file-handler/pull/118#issuecomment-1002716136
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

activescott commented:
I’d expect 2-3 that cover the statements that look similar to [this](https://github.com/activescott/serverless-aws-static-file-handler/blob/f03987ebc2ad7f18f0e12d6920b2f8e4da5d4032/src/test/StaticFileHandler.js#L66) or [this](https://github.com/activescott/serverless-aws-static-file-handler/blob/f03987ebc2ad7f18f0e12d6920b2f8e4da5d4032/src/test/StaticFileHandler.js#L96). 

You might want to add a ‘mockV2Event’ or something similar to the ‘mockEvent’ method at the top of that file. 

If you’re feeling taxed by this let me know and I can write something up tomorrow.

activescott commented:
I created a PR at tomchiverton/serverless-aws-static-file-handler#1 with an example and a update updates to this one.

tomchiverton commented:
I've added two unit tests with a mock as suggested, and updated to include the MR you sent to my branch.

CI tests pass now, apart from one that looks like it's set up wrong / will never run on branches ?

activescott commented:
shoot. I should have squashed this with a conventional commit 🤦‍♂️ . I'll have to come back and fix that later tonight or tomorrow...
